### ISSUE TITLE
Scheduled Queries

### ISSUE THREAD

**Issue by Ryxias** [airbnb/streamalert#1116]
## Background

This is just a kind of informal teaser to an upcoming product that I'm gonna open source, StreamQuery.

### Desired Change

We would like to be able to run scheduled SQL queries on Athena tables, and feed their results back into StreamAlert for rule analysis.

This allows StreamAlert to perform more stateful alerting.

### Architecture

![image](https://user-images.githubusercontent.com/13323701/74182491-07059380-4bf8-11ea-8824-554b8117b8b5.png)

**Comment by chunyong-lin** [comment]
SHIPIT
![good_luck](https://user-images.githubusercontent.com/21151436/74181557-6662a400-4bf6-11ea-8fe7-f8680350706e.gif)

**Comment by jack1902** [comment]
Whereabouts are the queries defined? Within the lambda in a `.sql` file which the cloudwatch event references or is the query on the event itself?

**Comment by blakemotl** [comment]
Currently they are defined in "query packs" which are python files where you construct a python object with the raw query and some other information, such as name and frequency.

**Comment by Ryxias** [comment]
Right now they look like this:

```
QueryPackConfiguration(
    name='athena_any_query',
    description='This query returns all Athena queries... how meta!',
    handler='athena:csirt',
    query="""
SELECT
  eventtime,
  json_extract(requestparameters['queryexecutioncontext'], '$.database') as database_name,
  requestparameters['querystring'] as querystring,
  useridentity['type'] as user_identity_type,
  useridentity['arn'] as user_identity_arn,
  dt
FROM
  "streamalert"."cloudwatch_cloudtrail"
WHERE
  dt = '{utcdatehour_minus1hour}'
  -- Only Events from Athena
  AND eventsource = 'athena.amazonaws.com'
  AND eventname = 'StartQueryExecution'
  -- Only on the CSIRT Prod account
  AND recipientaccountid = '1234567890987654321'
  -- Filter out noisy ALTER and SHOW queries. SHOW queries are commonly run in automation
  -- by API clients, and ALTER queries are run commonly by the Athena partition function.
  AND upper(substr(requestparameters['querystring'], 1, 5)) NOT IN ('ALTER', 'SHOW ')
""",
    params=['utcdatehour_minus1hour'],
    tags=['hourly', 'production']
)
© 2020 GitHub, Inc.
```

And the result that it sends to streamalert looks like this:

```
{
    "streamquery_schema_version": "0.0.7",
    "execution": {
    "console_link": "https://console.aws.amazon.com/athena/home?region=us-east-1#query/history/asdkfjasdf-asdfjkhasdfjh-awefkwahjef-awefaw,
      "handler": "athena:csirt",
      "name": "athena_any_query",
      "description": "This query returns all Athena queries... how meta!'",
      "query": "SELECT table_name, * FROM whatever blah ---",
      "data_scanned_in_bytes": 12345678,
      "execution_time_ms": 1234,
      "tags": [
        "hourly",
        "production"
      ]
    },
    "data": {
      "headers": [
        "eventtime",
        "database_name",
        "querystring",
        "user_identity_type",
        "user_identity_arn",
        "dt"
      ],
      "rows": [
        {
          "eventtime": "2019-10-09T18:04:59Z",
          "database_name": "streamalert",
          "querystring": "SELECT * FROM things",
          "user_identity_type": "AssumedRole",
          "user_identity_arn": "arn:aws:sts::2345678987654:assumed-role/airbnb_thing/airbnb_thing",
          "dt": "2019-09-10-11"
        }
      ],
      "count": 1
    }
  }
```

**Comment by Ryxias** [comment]
Likely changes in the result sent to StreamAlert:
* `handler` will no longer show up. It was reserved for a different use but is no longer necessary.
* `parameters` will be available which has a mapping of the query parameters to their calculated values.

**Comment by ryandeivert** [comment]
@Ryxias you can probably close this unless you want to wait until 3.1.0 is cut

**Comment by Ryxias** [comment]
This will be released in 3.1.0
